src/rsysreqs/scripts/scripts.go

```diff
@@ -0,0 +1,21 @@
+package scripts
+
+import (
+	"errors"
+
+	"rsysreqs/rules"
+)
+
+var ErrUnsupportedSystem = errors.New("unsupported system")
+
+type ScriptGenerator interface {
+	InstallScripts(packages []string) []string
+}
+
+func NewScriptGenerator(sys rules.System) (ScriptGenerator, error) {
+	switch {
+	case sys.Os == "linux" && sys.Distribution == "ubuntu":
+		return ubuntuScriptGenerator{}, nil
+	}
+	return nil, ErrUnsupportedSystem
+}
```

Comment:
Is this the right pattern/convention for a generic operation with many different implementations?

Reply:
This looks great to me!